consensus/state_sync/server.go

```diff
@@ -0,0 +1,125 @@
+package state_sync
+
+import (
+	"fmt"
+
+	typesCons "github.com/pokt-network/pocket/consensus/types"
+	"github.com/pokt-network/pocket/shared/codec"
+	coreTypes "github.com/pokt-network/pocket/shared/core/types"
+	cryptoPocket "github.com/pokt-network/pocket/shared/crypto"
+	"google.golang.org/protobuf/types/known/anypb"
+)
+
+// This module is responsible for handling requests and business logic that advertises and shares
+// local state metadata with other peers synching to the latest block.
+type StateSyncServerModule interface {
+	// Advertise (send) the local state sync metadata to the requesting peer
+	HandleStateSyncMetadataRequest(*typesCons.StateSyncMetadataRequest) error
+
+	// Send the block being requested by the peer
+	HandleGetBlockRequest(*typesCons.GetBlockRequest) error
+}
+
+func (m *stateSync) HandleStateSyncMetadataRequest(metadataReq *typesCons.StateSyncMetadataRequest) error {
+	serverNodePeerId, err := m.GetBus().GetConsensusModule().GetCurrentNodeAddressFromNodeId()
```

Comment:
1. `serverNodePeerId` will eventually be handeled by the P2P module
2. I feel like `peerId` could potentially be replaced by address?

Given the two points above, could we potentially just use the `address` instead? It would eliminate the need for `GetCurrentNodeAddressFromNodeId`. Wdyt?

Reply:
I am not sure if I completely addressed your comment, but, in order to remove peerId naming, and starting to explicitly work with addresses, I did the following:

1. I added `nodeAddress` to the consensus module, and added initialization for it. 
2. I updated state_sync.proto `peerId` fields to `peer_address`
3. I removed `GetCurrentNodeAddressFromNodeId`  but, since state sync module needs to access the current node id I put a getter function `GetNodeAddress` which returns the `nodeAddress` field of consensus module.


Lmk wdyt.